src/main/resources/GoogleApps/Groovy.groovy

```diff
@@ -365,27 +368,32 @@ public class GoogleAppsGroovy {
             }
         }
         String id = user.get("resourceName");
-        if(id.startsWith("people/") && id.length()>7) id = id.substring(7);
+        if(id.startsWith("people/") && id.length()>7) {
+            id = id.substring(7);
+        }
         def db = context.getDatabase()
         try {
             // Force main wiki database to create the user as global
             context.setDatabase("xwiki")
-            if(email == "")
-                email = (user.emailAddresses!=null && user.emailAddresses.size()>0) ? user.emailAddresses[0].value : "";
+            if(usersEmailAddress == "") {
+                usersEmailAddress = (user.emailAddresses!=null && user.emailAddresses.size()>0) ? user.emailAddresses[0].value : "";
+            }
             def wikiUserList = services.query.xwql("from doc.object(GoogleApps.GoogleAppsAuthClass) as auth where auth.id=:id").bindValue("id", id).execute()
             if ((wikiUserList==null) || (wikiUserList.size()==0))
-                wikiUserList = services.query.xwql("from doc.object(XWiki.XWikiUsers) as user where user.email=:email").bindValue("email", email).execute()
+                wikiUserList = services.query.xwql("from doc.object(XWiki.XWikiUsers) as user where user.email=:email").bindValue("email", usersEmailAddress).execute()
 
             if ((wikiUserList==null) || (wikiUserList.size()==0)) {
                 // user not found.. need to create new user
-                xwikiUser = email.substring(0, email.indexOf("@"));
+                xwikiUser = usersEmailAddress.substring(0, usersEmailAddress.indexOf("@"));
                 // make sure user is unique
                 xwikiUser = xwiki.getUniquePageName("XWiki", xwikiUser);
                 // create user
                 def parentref = xwiki.getDocumentAsAuthor("Main.UserDirectory").getDocumentReference()
                 def randomPassword = RandomStringUtils.randomAlphanumeric(8)
-                if(user.names==null || user.names.size()==0) throw new NullPointerException("Sorry, users without names are not supported.");
-                def isCreated = xwiki.getXWiki().createUser(xwikiUser, ["first_name" : user.names[0].givenName, "last_name" : user.names[0].familyName, "email" : email,  "password" : randomPassword], parentref, null, null, "edit", context.context)
+                if(user.names==null || user.names.size()==0) {
+                    throw new NullPointerException("Sorry, users without names are not supported.");
```

Comment:
You're trowing NPE knowing that you will catch and check it in other place? I can see that you're used to check for `null` everywhere, but I don't think is a good idea in trowing this here.
I would rather log something here.

Reply:
I think thtat the intent of throwing the NPE here is to transmit a message to the user. Any exception would probably do.
Should I change it to a java.lang.Exception ?

Reply:
The idea is to warn with a nice message in the UI and not to break the page with an error. Again would be acceptable for this fix, but we're doing a lot of compromises in rush which is not fine at all. Keep it like that for now.